be/src/vec/core/block.cpp

```diff
@@ -151,6 +151,13 @@ void Block::insert(ColumnWithTypeAndName&& elem) {
     data.emplace_back(std::move(elem));
 }
 
+void Block::insert_and_resize(ColumnWithTypeAndName&& elem) {
+    index_by_name.emplace(elem.name, data.size());
+    // usualy elem.column always is const column, so we just clone it.
+    elem.column = elem.column->clone_resized(std::max(rows(), size_t(1)));
```

Comment:
1. `clone_resized` will cause some data loss
2. `clone_resized` will cause the `memcpy, new, delete`. 
I don't think it should be a member function